test/libsolidity/syntaxTests/storageLayoutSpecifier/layout_specification_binary_expression.sol

```diff
@@ -0,0 +1,4 @@
+contract C layout at 0xffff * (0x123 + 0xABC) { }
+// ====
+// stopAfter: parsing
```

Comment:
Is this even needed here? There's no type checking so it won't even fail, will it?

Reply:
Actually, just realized why this was bothering me. The PR only implements parsing but does not disable code generation. We shouldn't be generating the code for `layout at `0` when user specifies something else so this should be generating an error until codegen is implemented. Then having these here to prevent running into that error would make sense.

So overall, In this PR we should refuse to generate code and show storage layout output when `layout at` is specified. In the typechecking PR, where we'll know the value, we can relax it to cases where base is non-zero. And in the codegen PR we should be able to lift the restriction completely.